homeassistant/components/device_tracker/unifi.py

```diff
@@ -81,7 +89,12 @@ def _update(self):
             _LOGGER.error("Failed to scan clients: %s", ex)
             clients = []
 
-        self._clients = {client['mac']: client for client in clients}
+        self._clients = {
+            client['mac']: client
+            for client in clients
+            if (dt_util.utcnow() - dt_util.utc_from_timestamp(float(
+                client['last_seen']))) < self._detection_time
+            }
```

Comment:
The dict comprehension gets a lot clearer if you do the following:

new_enough = dt_util.utcnow() - self._detection_time

self._clients = {c['mac']: c for c in clients if c['last_seen'] > new_enough}

And I think keeps you under 80 chars